presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.teradata.tempto.AfterTestWithContext;
+import com.teradata.tempto.BeforeTestWithContext;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.query.QueryExecutionException;
+import io.airlift.http.client.HttpClient;
+import io.airlift.http.client.HttpClientConfig;
+import io.airlift.http.client.HttpStatus;
+import io.airlift.http.client.Response;
+import io.airlift.http.client.jetty.JettyHttpClient;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.facebook.presto.tests.TestGroups.CANCEL_QUERY;
+import static com.google.common.util.concurrent.Uninterruptibles.sleepUninterruptibly;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static java.lang.String.format;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.fail;
+
+public class SqlCancelTests
+        extends ProductTest
+{
+    private static final String TABLE_NAME = "cancel_query_table_name";
+    private static final long WAIT_TIME = 10000;
+    private static final long STEP = 100;
+
+    private static ExecutorService executor;
+    private static HttpClient httpClient;
```

Comment:
This is only used in QueryCanceller, no reason to make it a class variable.
I guess you do it so that you can close it, but you can instead have QueryCanceller implement Closeable and call QueryCanceller.close().  For that matter you can create the client in the QueryCanceller constructor too instead of doing it in set up and passing it in.